opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -172,9 +172,118 @@ message MetricDescriptor {
   }
   Type type = 4;
 
+  // Temporality is the temporal quality values of a metric have. It
+  // describes how those values relate to the time interval over which they
+  // are reported.
+  enum Temporality {
+    // TEMPORALITY_INVALID is the default Temporality, it MUST not be
+    // used.
+    TEMPORALITY_INVALID = 0;
```

Comment:
UNKNOWN (or just INVALID temporality is included in the generated name, this will be TEMPORALITY_TEMPORALITY_INVALID) is usually the recommended name for this by the proto3 conventions https://developers.google.com/protocol-buffers/docs/proto3#enum

Reply:
I would probably go with just "INVALID"

Reply:
The problem is collision with other enums if they are not prefixed. Both `Type` and `Temporality` would have an `INVALID` and that would cause a collision. The same for `UNSPECIFIED`.

I originally used synonyms, but then there is an unimplemented semantic difference.

Would a suffix work better for you?

Reply:
There is no collision because the name of the enum is used as prefix for all the values.

Reply:
I think you'll need to take that up with `protoc` :wink:  :

```
$ make gen-go
...
protoc --go_out=plugins=grpc:/home/tyler/go/src opentelemetry/proto/metrics/v1/metrics.proto
opentelemetry/proto/metrics/v1/metrics.proto:159:5: "INVALID" is already defined in "opentelemetry.proto.metrics.v1.MetricDescriptor".
opentelemetry/proto/metrics/v1/metrics.proto:159:5: Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it.  Therefore, "INVALID" must be unique within "opentelemetry.proto.metrics.v1.MetricDescriptor", not just within "Temporality".
make: *** [makefile:25: gen-go] Error 1
```

Reply:
You are correct, I've just saw that in the go generated code, this is ugly because in java this will be:

```java
class MetricDescriptor {
enum Temporality {
  TEMPORALITY_INVALID = 0,
}
}

// And usage
Temporality temporality = Temporality.TEMPORALITY_INVALID
```

Reply:
Another option is to have Temporality defined not a child of MetricDescriptor but as a standalone enum MetricDescriptorTemporality for example. This is just an idea, let me know what you think?

Reply:
Yeah, not great. :cry:

Would a suffix be better?

I like the idea of consistently using the `INVALID` (and `UNSPECIFIED`) term to convey the same meaning (i.e. don't use or there is no defined refinement).

I'm open to suggestions.

Reply:
> Another option is to have Temporality defined not a child of MetricDescriptor but as a standalone enum MetricDescriptorTemporality for example. This is just an idea, let me know what you think?

I was thinking about moving to a different scope, but seems like that would only propagate the problem. Like, why is Temporality in the top scope but not Type? And then it would become less clear when resolving the collision for the Monotonic and Domain defaults of UNSPECIFIED. Which one of those should be in the top scope? What happens if we add another refinement? :shru

https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors#metrickind

Reply:
Do you mean suffix just for `INVALID` or for all entries in the enum? I kind of like more `INVALID_TYPE` :)

Reply:
Yeah, just for `INVALID_*` (and `UNSPECIFIED_*`). I'll switch to that.